agent/engine/containermetadata/acquire_metadata.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package containermetadata
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/config"
+	docker "github.com/fsouza/go-dockerclient"
+)
+
+// acquireNetworkMetadata parses the docker.NetworkSettings struct and
+// packages the desired metadata for JSON marshaling
+func acquireNetworkMetadata(settings *docker.NetworkSettings) NetworkMetadata {
+	if settings == nil {
+		return NetworkMetadata{}
+	}
+
+	// Get Port bindings from docker settings
+	ports, err := api.PortBindingFromDockerPortBinding(settings.Ports)
+	if err != nil {
+		ports = nil
+	}
+
+	// This metadata is available in two different places in NetworkSettings
+	gateway := settings.Gateway
+	iPAddress := settings.IPAddress
+	iPv6Gateway := settings.IPv6Gateway
+
+	// Assume there is at most one network mode (And if none, network is "none")
+	networkModeFromContainer := ""
+	if len(settings.Networks) == 1 {
+		for modeFromSettings, containerNetwork := range settings.Networks {
+			networkModeFromContainer = modeFromSettings
+			gateway = containerNetwork.Gateway
+			iPAddress = containerNetwork.IPAddress
+			iPv6Gateway = containerNetwork.IPv6Gateway
+		}
+	} else if len(settings.Networks) == 0 {
+		networkModeFromContainer = "none"
```

Comment:
I believe Docker has 4 network modes, bridge, host, mapped, and none. mapped mode is not supported by ECS I think. If this is not 1, I assume it will be "none" mode by default which seems to be the case in my testing. I have not done any testing of different network modes yet.